streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamPrint.java

```diff
@@ -62,13 +62,19 @@ public void init(ProcessorContext context) {
             if (valueSerde == null) {
                 this.valueSerde = context.valueSerde();
             }
-        }
 
-        @Override
-        public void process(final K key, final V value) {
-            final K deKey = (K) maybeDeserialize(key, keySerde.deserializer());
-            final V deValue = (V) maybeDeserialize(value, valueSerde.deserializer());
-            action.apply(deKey, deValue);
+            // no mapper provided, using a default one which takes care of checking byte[] for deserialization
+            if (mapper == null) {
+                mapper = new KeyValueMapper<K, V, String>() {
+                    @Override
+                    public String apply(K key, V value) {
+                        final K deKey = (K) maybeDeserialize(key, keySerde.deserializer());
```

Comment:
Thanks @dguy it was just what I was thinking while trying the test suggested by @guozhangwang. So it seems that the serde parameters are useless. We don't need to check that the k/v are byte[ ] or some K and V types because in any case the serdes will be of the same type accordingly. Finally the doubt, why we have today in the current code the needs for checking with "instance of" if k/v are byte[ ] and doing a deserialization if we are saying that it should never happen ?